src/rosdep2/main.py

```diff
@@ -461,6 +469,20 @@ def _package_args_handler(command, parser, options, args):
                 print("Skipping non-existent path " + path)
         set_workspace_packages(ws_pkgs)
 
+    # Print installed conflicting or replaced packages
+    if command in ['install', 'check'] and (options.ignore_src or options.from_paths):
+        workspace_pkgs = get_workspace_packages()
+        conflicted_pkgs = get_conflicted_packages()
+        replaced_pkgs = get_replaced_packages()
+
+        for package in workspace_pkgs:
+            if package in conflicted_pkgs:
+                print("WARNING: conflicting installed packages: '{0}' is conflicting '{1}'!".format(
+                    conflicted_pkgs[package], package))
+            if package in replaced_pkgs:
+                print("WARNING: '{0}' replaces '{1}', but '{1}' is already installed!".format(
+                    replaced_pkgs[package], package))
+
```

Comment:
You are only checking if the rosdep keys are in the workspace. What about rosdep keys marked as conflicting that are not in your workspace? You are not checking if those are installed.